Snowflake.Data.Tests/SFDbDataReaderIT.cs

```diff
@@ -1323,5 +1323,59 @@ public void TestResultSetMetadata()
                 conn.Close();
             }
         }
+        
+        [Test]
+        public void TestHasRows()
+        {
+            using (DbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString;
+                conn.Open();
+
+                DbCommand cmd = conn.CreateCommand();
+                cmd.CommandText = "select 1 where 1=2";
+                
+                DbDataReader reader = cmd.ExecuteReader();
+                Assert.IsFalse(reader.HasRows);
+                reader.Close();
+
+                cmd.CommandText = "select 1;" +

```

Comment:
I would maybe split it into separate TestMultiStatementHasRows

Reply:
Splitted